release/nightly_tests/setup_cluster_compute_config_updater.py

```diff
@@ -0,0 +1,147 @@
+import os
+import ray
+import asyncio
+import logging
+from typing import List, Union
+from dataclasses import dataclass
+import requests
+import anyscale
+import time
+import argparse
+from ray.util.scheduling_strategies import NodeAffinitySchedulingStrategy
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ClusterComputeConfigUpdate:
+    # Path to the field that needs to be updated
+    field_path: List[Union[str, int]]
+    # Field value to be updated to
+    field_value: Union[str, int]
+    # When to update the field in terms of seconds since run of the updater
+    update_time_seconds_since_run: int
+
+    def __init__(self, update: str):
+        field_path, field_value, update_time_seconds_since_run = update.split(":")
+
+        self.field_path = []
+        for field in field_path.split("."):
+            if field.isdigit():
+                self.field_path.append(int(field))
+            else:
+                self.field_path.append(field)
+
+        if field_value.isdigit():
+            self.field_value = int(field_value)
+        else:
+            self.field_value = field_value
+
+        self.update_time_seconds_since_run = int(update_time_seconds_since_run)
+
+
+class ClusterComputeConfig:
+    def __init__(self):
+        self.cluster_id = os.environ["ANYSCALE_CLUSTER_ID"]
+        sdk = anyscale.AnyscaleSDK()
+        self.cluster = sdk.get_cluster(self.cluster_id)
+        self.compute_config_dict = anyscale.compute_config.get(
+            name="", _id=self.cluster.result.cluster_compute_id
+        ).config.to_dict()
+        logging.info(
+            f"Fetched compute config {self.compute_config_dict} "
+            f"for cluster {self.cluster}"
+        )
+
+    def update(self, field_path: List[Union[str, int]], field_value: Union[str, int]):
+        current = self.compute_config_dict
+        for field in field_path[:-1]:
+            current = current[field]
+        current[field_path[-1]] = field_value
+
+        new_compute_config = anyscale.compute_config.models.ComputeConfig.from_dict(
+            self.compute_config_dict
+        )
+        new_compute_config_name = anyscale.compute_config.create(
+            new_compute_config, name=None
+        )
+        new_compute_config_id = anyscale.compute_config.get(
+            name=new_compute_config_name
+        ).id
+
+        response = requests.put(
+            f"https://console.anyscale-staging.com/api/v2/sessions/{self.cluster_id}/"
+            "cluster_config_with_session_idle_timeout",
+            params={
+                "build_id": self.cluster.result.cluster_environment_build_id,
+                "compute_template_id": new_compute_config_id,
+            },
+            headers={"Authorization": f"Bearer {os.environ['ANYSCALE_CLI_TOKEN']}"},
```

Comment:
For my own understanding, is `ANYSCALE_CLI_TOKEN` always available in our release tests?

Reply:
Yes as long as we run the release test on Anyscale.